File: src/features/typing/data/go/container_with_most_water.go
package main

func maxArea(height []int) int {
	left, right := 0, len(height)-1
	best := 0
	for left < right {
		w := right - left
		h := height[left]
		if height[right] < h {
			h = height[right]
		}
		if w*h > best {
			best = w * h
		}
		if height[left] < height[right] {
			left++
		} else {
			right--
		}
	}
	return best
}

File: src/features/typing/data/ts/container_with_most_water.ts
function maxArea(height: number[]): number {
  let left = 0;
  let right = height.length - 1;
  let best = 0;
  while (left < right) {
    const w = right - left;
    const h = Math.min(height[left], height[right]);
    best = Math.max(best, w * h);
    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
  }
  return best;
}
